psi4/src/psi4/libmints/wavefunction.cc

```diff
@@ -804,16 +812,16 @@ void Wavefunction::force_soccpi(const Dimension &soccpi) {
             "Wavefunction::force_soccpi: Cannot set soccpi since alpha and beta densities must be the same for this "
             "Wavefunction.");
     }
+    auto docc = doccpi();
     for (int h = 0; h < nirrep_; h++) {
-        if ((soccpi[h] + doccpi_[h]) > nmopi_[h]) {
+        if ((soccpi[h] + doccpi()[h]) > nmopi_[h]) {
             throw PSIEXCEPTION(
                 "Wavefunction::force_soccpi: Number of doubly and singly occupied orbitals in an irrep cannot exceed "
                 "the total number of molecular orbitals.");
         }
-        soccpi_[h] = soccpi[h];
-        nalphapi_[h] = doccpi_[h] + soccpi_[h];
+        nalphapi_[h] = docc[h] + soccpi[h];
```

Comment:
What's the idea with `docc`, `doccpi()`, and `doccpi_`? Wh yso many variables that assumedly are the same?

Reply:
I can change l. 817 to use `docc` instead of `doccpi()`. The point is to avoid recompute.

Reply:
What is the difference between `docc`, `doccpi()` and `doccpi_`?

Reply:
Shouldn't `soccpi_` and `doccpi_` be removed?

Reply:
I'm legitimately confused. `soccpi_` and `doccpi_` _are_ removed.

Like I said previously, the point of `docc` is to avoid recompute of `doccpi()` every iteration of the for loop.

Reply:
```suggestion
        nalphapi_[h] = docc[h] + input_socc[h];
```

Reply:
You can add Dimension objects directly rather than doing it by irrep for further simplifications, but agreed that the function can and should be simplified now.

Reply:
I was legitimately confused because the array names were inconsistent. The point of these functions is that an input array is passed in and then combined with the computed arrays. I have proposed above to use a more consistent naming scheme: `socc` and `docc` for the precomputed arrays and `input_socc` and `input_docc` for the overriding inputs.